modules/ADE/private/KeyVault/Deploy-AzureKeyVault.ps1

```diff
@@ -6,15 +6,21 @@ function Deploy-AzureKeyVault {
         [object] $wildcardCertificatePath
     )
 
-    Deploy-ArmTemplate 'Azure Key Vault' $armParameters -resourceGroupName $armParameters.keyVaultResourceGroupName -bicep
-
-    Write-Status "Configuring Azure Key Vault $($armParameters.keyVaultName)"
-
-    Set-AzureKeyVaultSecret $armParameters.keyVaultName 'resourcePassword' $secureResourcePassword
-    Deploy-WildcardCertificateToAzureKeyVault $armParameters.keyVaultName $secureCertificatePassword $wildcardCertificatePath
-    New-AzureKeyVaultKey $armParameters.keyVaultName 'containerRegistry'
-    
-    Set-AzureKeyVaultResourceId $armParameters
-
+    # Similar to other 'Deploy' functions, skip this KV Creation if the KV already exists
+    If ($armParameters.keyVaultResourceID){ # We already Confirmed that this Azure resource existed earlier in script Set-InitialArmParameters.ps1 and populated the property keyVaultResourceID
```

Comment:
It's ok to skip the soft delete check, but don't skip the creation. Since ARM templates can be incremental, we may still want to do the deployment if we changed any properties of the Key Vault (e.g. added new keys, RBAC, etc). I think you could just do `if`/`else` by renaming your current `else if` to `else`, and unwrapping your current `else`… if that makes sense.